pkg/agent/baker.go

```diff
@@ -545,6 +547,9 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 		"IsIPv6DualStackFeatureEnabled": func() bool {
 			return cs.Properties.FeatureFlags.IsFeatureEnabled("EnableIPv6DualStack")
 		},
+		"IsAzureCNIOverlayFeatureEnabled": func() bool {
+			return cs.Properties.OrchestratorProfile.KubernetesConfig.IsUsingNetworkPluginMode("overlay")
```

Comment:
do we need to return false if some property is nil?

Reply:
can cs, Properties, OrchestratorProfile, or KubernetesConfig be null? It doesn't look like it's being checked in the other parts, but I can add that here

Reply:
It seems like that only some funcs check it. I am fine with current code since most funcs do not check it and we do not hit any troubles.

Reply:
https://github.com/Azure/AgentBaker/blob/542d2dc3e603f7f01d8b34757c2c0d4855cdbe21/pkg/agent/baker.go#L440-L443